impeller/aiks/experimental_canvas.cc

```diff
@@ -0,0 +1,244 @@
+
+  auto inline_pass = std::make_unique<InlinePassContext>(
+      renderer_, *entity_pass_targets_.back(), 0);
+  inline_pass_contexts_.emplace_back(std::move(inline_pass));
+
+  auto result = inline_pass_contexts_.back()->GetRenderPass(0u);
+  render_passes_.push_back(result.pass);
+}
+
+bool ExperimentalCanvas::Restore() {
+  FML_DCHECK(transform_stack_.size() > 0);
+  if (transform_stack_.size() == 1) {
+    return false;
+  }
+
+  if (transform_stack_.back().rendering_mode ==
+      Entity::RenderingMode::kSubpass) {
+    auto inline_pass = std::move(inline_pass_contexts_.back());
+
+    SaveLayerState save_layer_state = save_layer_state_.back();
+    save_layer_state_.pop_back();
+
+    std::shared_ptr<Contents> contents =
+        PaintPassDelegate(save_layer_state.paint)
+            .CreateContentsForSubpassTarget(inline_pass->GetTexture(),
+                                            transform_stack_.back().transform);
+
+    inline_pass->EndPass();
+    render_passes_.pop_back();
+    inline_pass_contexts_.pop_back();
+
+    Entity element_entity;
+    element_entity.SetClipDepth(GetClipHeight());
+    element_entity.SetContents(std::move(contents));
+    element_entity.SetBlendMode(save_layer_state.paint.blend_mode);
+    element_entity.SetTransform(Matrix::MakeTranslation(
+        Vector3(save_layer_state.coverage.GetOrigin())));
+    element_entity.Render(renderer_, *render_passes_.back());
+  }
+
+  transform_stack_.pop_back();
+
+  return true;
+}
+
+void ExperimentalCanvas::DrawTextFrame(
+    const std::shared_ptr<TextFrame>& text_frame,
+    Point position,
+    const Paint& paint) {
+  Entity entity;
+  entity.SetClipDepth(GetClipHeight());
+  entity.SetBlendMode(paint.blend_mode);
+
+  auto text_contents = std::make_shared<TextContents>();
+  text_contents->SetTextFrame(text_frame);
+  text_contents->SetColor(paint.color);
+  text_contents->SetForceTextColor(paint.mask_blur_descriptor.has_value());
+  text_contents->SetScale(GetCurrentTransform().GetMaxBasisLengthXY());
+
+  entity.SetTransform(GetCurrentTransform() *
+                      Matrix::MakeTranslation(position));
+
+  // TODO(bdero): This mask blur application is a hack. It will always wind up
+  //              doing a gaussian blur that affects the color source itself
+  //              instead of just the mask. The color filter text support
+  //              needs to be reworked in order to interact correctly with
+  //              mask filters.
+  //              https://github.com/flutter/flutter/issues/133297
+  entity.SetContents(
+      paint.WithFilters(paint.WithMaskBlur(std::move(text_contents), true)));
+
+  AddEntityToCurrentPass(std::move(entity));
+}
+
+void ExperimentalCanvas::AddEntityToCurrentPass(Entity entity) {
+  auto transform = entity.GetTransform();
+  entity.SetTransform(
+      Matrix::MakeTranslation(Vector3(-GetGlobalPassPosition())) * transform);
+  entity.Render(renderer_, *render_passes_.back());
```

Comment:
Is the ExperimentalCanvas using the old clipping stencil buffer implementation? (the reason I ask is it isn't setting the "NewClipDepth") I didn't think that was still in there...